src/Di/Inject.php

```diff
@@ -20,4 +20,12 @@
      * @var bool
      */
     public $optional = false;
```

Comment:
There does not seem to be anything preventing it from being private, other than backwards compatibility